### PR TITLE
Simplify build processes

### LINKED ISSUE #8: Workflow and current build scripts are too complicated
Current workflows and build scripts are too complicated for newcomers and even, for returning users. We should try to simplify these processes. Preferably, move the scripts to a `scripts` folder and update README.md with the new flows.

### PR DESCRIPTION
**Pull request by nguyenlamlll**
In short, this resolves #8.

This pull request introduces some major changes to the build processes of this AdminGui plugin, plus some big changes on the main README.md file.

- The available commands will become `init`, `run`, and `publish`. To use them using PowerShell from the root directory of this repository, simply invoke them by adding `.\` to the front, e.g., `.\run`.
---
`init` command will initialize and create all required prerequisites for a developer to begin working on this repository. 

`run` command builds the plugins, AdminGui, create debug packages out of them, start an IISExpress server, and remove those debug packages. This is for debugging sessions on local dev machines. 

`publish` command will build the plugins and AdminGui then create Release Nuget packages that are ready for production.

---

- README.md is re-written almost entirely. There are still many "Coming soon" parts but they are not really in the scope of this PR (which is to solve #8, again). So, I will try to update them with another PR 😄. However, most importantly, the file is updated with basic instructions on how to build from source and contributing (aka using the commands above) to the repository.

The scripts are test on my machine and they are working fine. But they can get crazy over different environments... If it's possible, please help me test them with fresh clones on your machines. Thanks!

### CODE REVIEW

File: Tools/AdminGuiBuildCore.psm1 [hunk 1]
```diff
@@ -89,7 +89,6 @@ function Build-Plugins($buildConfiguration = "Debug") {
     Push-Location .\Angular2ModelGenerator
     
     & $script:BuildToolPath AdminGuiPlugin.sln /t:rebuild /p:configuration=$buildConfiguration
-    #dotnet build AdminGuiPlugin.sln --configuration $buildConfiguration
```
Comment (nguyenquanghuy031295) [inline Tools/AdminGuiBuildCore.psm1]: Remove this comment for dotnet build.

**Comment by bantolov** [comment]
I am getting some errors in .\run. Please review the attached console output [run output.txt](https://github.com/Rhetos/AdminGui/files/2424353/run.output.txt). It includes the commands that are executed and the command output.
The problem might be trivial but i don't have much experience with npm.

**Comment by ngoquoc** [comment]
> I am getting some errors in .\run. Please review the attached console output [run output.txt](https://github.com/Rhetos/AdminGui/files/2424353/run.output.txt). It includes the commands that are executed and the command output.
> The problem might be trivial but i don't have much experience with npm.

It seems that you don't have `gulp` installed globally. Can you try this command:
`npm install gulp -g`
Then retry running `.\run`.

Nevertheless, I think it should be better to use `gulp` binary from local `node_modules` rather than the global one. The local gulp should be declared as one of `devDependencies` in `package.json`. 
@nguyenlamlll @nguyenquanghuy031295 What do you think?

**Comment by huynguyen-95** [comment]
@ngoquoc 
Actually in package.json ```devDependencies``` I added ```gulp``` version 3.9.1

**Comment by nguyenquanghuy031295** [comment]
@bantolov @ngoquoc 
Could you check again? I just push some changes.

**Comment by nguyenquanghuy031295** [comment]
@bantolov @ngoquoc
Could you check again? I just push some changes.

**Comment by nguyenlamlll** [comment]
Yes, `gulp` is already installed within `package.json` but the problem lies in the way we call it. Solved it.
And as I have just updated from `master`, there are still some more problems with the latest changes. I will work on this and ping you guys again when things are ready to go.

**Comment by nguyenlamlll** [comment]
Hi @bantolov, @ngoquoc, the scripts seem better now (and they can be run on AppVeyor agent). Can you guys help me review them? Thanks.

**Comment by ngoquoc** [comment]
> Hi @bantolov, @ngoquoc, the scripts seem better now (and they can be run on AppVeyor agent). Can you guys help me review them? Thanks.

Great news, thanks! I will review it soon.

**Comment by bantolov** [comment]
> Hi @bantolov, @ngoquoc, the scripts seem better now (and they can be run on AppVeyor agent). Can you guys help me review them? Thanks.

The build works well on my machine now!

**Comment by ngoquoc** [comment]
@nguyenlamlll I got this error after running `.\Run.ps1`
```
[Error] DeployPackages: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\qngo\Documents\Projects\AdminGUI\github-repo\rhetos\AdminGui\2CS.RhetosBuild\Rhetos\Resources\AdminGuiCompile/dist/admingui.js'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalCopy(String sourceFileName, String destFileName, Boolean overwrite, Boolean checkHost)
   at Angular2ModelGenerator.Angular2ModelGenerator.CopyCompiledFile()
   at Angular2ModelGenerator.Angular2ModelGenerator.Generate()
   at Rhetos.Deployment.ApplicationGenerator.ExecuteGenerators(Boolean deployDatabaseOnly) in C:\My Projects\Rhetos\Source\Rhetos.Deployment\ApplicationGenerator.cs:line 99
   at DeployPackages.Program.GenerateApplication(ILogger logger, Arguments arguments) in C:\My Projects\Rhetos\Source\DeployPackages\Program.cs:line 180
   at DeployPackages.Program.Main(String[] args) in C:\My Projects\Rhetos\Source\DeployPackages\Program.cs:line 68

=== ERROR SUMMARY ===
DirectoryNotFoundException: Could not find a part of the path 'C:\Users\qngo\Documents\Projects\AdminGUI\github-repo\rhetos\AdminGui\2CS.RhetosBuild\Rhetos\Resources\AdminGuiCompile/dist/admingui.js'.
```

The Rhetos server is successfully up though, but it showed blank page when I tried accessing http://localhost:9000/Resources/AdminGui/

**Comment by nguyenlamlll** [comment]
@ngoquoc: 
The problem you are facing right now is caused by the way `tsc` command is called inside the plugin ([this line](https://github.com/Rhetos/AdminGui/blob/ecafc6c6ea30f8a439a6f69f0532fadcc6b26e78/RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.cs#L78)). By that, the plugin implies that we have global `tsc`. This is not a good approach but I think this is out of the scope of this PR. So... for now, we are required to install `tsc` globally.

**Comment by huynguyen-95** [comment]
@ngoquoc , @nguyenlamlll is right.
As we discuss in **refactor plugin**'s PR, the code using ```tsc``` for compiling should be refactored. I asked Tan about it and found the way to refactor.
Your error is the issue #18 I created.

**Comment by ngoquoc** [comment]
Thank you guys, I managed to get it running after installing typescript globally. I think this PR can be completed.

**Pull request merged**